frontend/src/components/trips/AddTrip.js

```diff
@@ -0,0 +1,197 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Formik, Field } from 'formik';
+import {
+  FieldFactory,
+  PrimaryButton,
+  Label,
+} from '../UI';
+import Notification from '../Notification';
+
+const StyledInput = FieldFactory(Field);
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  background-color: ${(props) => props.theme.colors.fill};
+  padding: 1rem;
+  margin: 30% auto;
+  margin-top:-1%;
+  width: 80%;
+  max-width: 400px;
+  border-radius: 10px;
+  -webkit-box-shadow: -10px 10px 40px 0px rgba(10,10,10,0.75);
+  -moz-box-shadow: -10px 10px 40px 0px rgba(10,10,10,0.75);
+  box-shadow: -10px 10px 40px 0px rgba(10,10,10,0.75);
+`;
+
+const StyledTextRow = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin-top: 0.75rem;
+
+  &:nth-child(1) {
+    margin: 0;
+  }
+`;
+
+const AddNotification = styled.div`
+width:80%;
+margin: 30% auto;
+margin-top:-27%;
+
+`;
+
+const StyledSelectRow = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 0.75rem;
+`;
+
+const StyledSelectColumn = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+
+  & input::-webkit-calendar-picker-indicator {
+    display: none;
+  }
+
+  &:nth-child(1) {
+    margin-right: 1rem;
+  }
+
+  &:nth-child(2) {
+    margin-left: 1rem;
+    
+  }
+  
+`;
+
+const StyledButton = styled(PrimaryButton)`
+  padding: 0.75rem;
+  height: 80%;
+  width: 100%;
+  margin-top: 0.75rem;
+
+  &:nth-of-type(1) {
+    margin-right: 0.75rem;
+  }
+
+  &:nth-of-type(2) {
+    margin-left: 0.75rem;
+  }
+`;
+
+const AddTrip = ({ closeAdd }) => {
+  const [notification, setNotification] = useState(null);
+  const onSubmit = (values) => {
+    const newvalues = {
+
+      startLocation: values.startLocation,
+      destination: values.destination,
+      seatsAvailable: values.seatsAvailable,
+      startTime: `${values.startTime} ${values.time}`,
+      price: values.price,
+      description: values.description,
+    };
+
+    const showNotification = (msg, color, seconds) => {
+      setNotification({ msg, color });
+      setTimeout(() => {
+        setNotification(null);
+      }, seconds * 1000);
+    };
```

Comment:
Denna ligger i App.js, du måste skicka med den som en property: https://reactjs.org/docs/components-and-props.html